src/services/CollectionArgumentsCreator.ts

```diff
@@ -1,57 +1,123 @@
-            aggs[caption || type + idx] = {
-                terms: {
-                    field: `${paths[0]}.raw`
-                }
-            };
+interface Agg {
+    filter: EsQuery | {};
+    aggs: {
+        name: {
+            terms: {
+                field: string;
+            }
         }
-    });
+    };
+}
 
-    return aggs;
-};
+interface ElasticSearchParams {
+    aggs: Aggs;
+    post_filter?: EsQuery;
+}
 
-const createPostFilterString = (search: string): string => {
-    const searchString = search ? decode(search) : null;
+const doubleStringify = (obj: {}): string => JSON.stringify(JSON.stringify(obj));
 
-    if (searchString && searchString.length > 0) {
-        return searchString;
-    }
-    return '';
-};
+/** filter aggregations that match the path out of the search object and return a new instance
+ *
+ * @param {EsQuery} searchObj
+ * @param {string} field
+ * @returns {EsQuery}
+ */
+const setFilteredSearchObj = (searchObj: EsQuery, field: string): EsQuery => {
+    const filteredSearchObj: EsQuery = { bool: { must: [] } };
 
-const doubleStringify = (obj: {}): string => JSON.stringify(JSON.stringify(obj));
+    searchObj.bool.must.forEach(
+        (obj: EsMatches) => {
+            if (obj.hasOwnProperty('query_string')) {
+                filteredSearchObj.bool.must.push(obj);
 
-const elasticQueryStringCreator = (indexConfig: IndexConfig, search: string): string => {
-    const obj: { aggs: {}, post_filter?: {} } = {
-        aggs: createAggsString(indexConfig.facet)
-    };
+            } else if (obj.bool && obj.bool.should && obj.bool.should.length > 0 && obj.bool.should[0].match) {
+                const { match } = obj.bool.should[0];
+                let keyIsField: boolean = false;
 
-    const postFilter = createPostFilterString(search);
+                for (let key in match) {
+                    if (match.hasOwnProperty(key)) {
+                        if (key === field) {
+                            keyIsField = true;
+                        }
+                    }
+                }
 
-    if (postFilter.length > 0) {
-        obj.post_filter = postFilter;
-    }
+                if (!keyIsField) {
+                    filteredSearchObj.bool.must.push(obj);
+                }
+            }
+        }
+    );
 
-    return `elasticsearch: ${doubleStringify(obj)}`;
+    return filteredSearchObj;
 };
 
-const cursorStringCreator = (cursor: string): string => {
-    return cursor ? `, cursor: ${JSON.stringify(cursor)}` : '';
+/** create a string with aggregations. In case of a querystring add a filter for each aggregation to optimize the counts of filters to be shown
+ *
+ * @param {FacetConfig[]} facets
+ * @param {EsQuery} searchObj
+ * @returns {Aggs}
+ */
+const createAggsString = (facets: FacetConfig[], searchObj: EsQuery | null): Aggs => {
+    const aggregations: Aggs = {};
+
+    facets.forEach(({ paths, caption, type }: FacetConfig, idx: number) => {
```

Comment:
Laten we [for...in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) aanhouden voor loops door arrays heen.
Als je de index nodig hebt zoals hier kan je `.entries()` gebriuken.
```
for (const [idx: number, {paths, caption, type}: FacetConfig] of facets.entries())
```